### PR TITLE
Test Kokkos CMake overhaul

### PR DESCRIPTION
**Pull request by masterleinad**
This is to test https://github.com/kokkos/kokkos/pull/2104.

### CODE REVIEW

**Comment by aprokop** [comment]
I think we should test the overhaul against not carrying our own `FindKokkos.cmake`.

File: docker/Dockerfile [hunk 1]
```diff
@@ -109,16 +109,20 @@ RUN SCRATCH_DIR=/scratch && mkdir -p ${SCRATCH_DIR} && cd ${SCRATCH_DIR} && \
 
 # Install Kokkos
 ARG KOKKOS_VERSION=2.8.00
-ARG KOKKOS_OPTIONS="--cxxstandard=c++14 --with-serial --with-openmp --with-options=disable_deprecated_code --with-cuda --with-cuda-options=enable_lambda"
```
Comment (dalg24) [inline docker/Dockerfile]: Keep and pass to `cmake`

Comment (dalg24) [reply docker/Dockerfile]: Passing `nvcc_wrapper` as the `CMAKE_CXX_COMPILER` is going to be a pain.  I recommend omitting the path and prepending `<kokkos source dir>/bin` to the `PATH` environment variable in the `RUN` block below.

**Comment by masterleinad** [comment]
Superseded by #73.

**Pull request closed without merging**